relayer/relayer.go

```diff
@@ -438,13 +448,32 @@ func (r *Relayer) RelayMessage(warpLogInfo *vmtypes.WarpLogInfo) error {
 	// Increment the request ID for the next message relay request
 	r.currentRequestID++
 
+	if !storeProcessedHeight {
+		return nil
+	}
+
 	// Update the database with the latest processed block height
-	err = r.db.Put(r.sourceBlockchainID, []byte(database.LatestProcessedBlockKey), []byte(strconv.FormatUint(warpLogInfo.BlockNumber, 10)))
-	if err != nil {
+	// First, check that the stored height is less than the current block height
+	// This is necessary because the relayer may be processing blocks out of order on startup
+	latestProcessedBlock, err := r.getLatestProcessedBlockHeight()
```

Comment:
Maybe not for this PR, but could you create an issue to track the latest processed block height in memory so that we can get rid of the unnecessary disk reads here?